clients/privacy-center/packages/fides-consent/src/lib/consent-banner.ts

```diff
@@ -0,0 +1,553 @@
+  flex: auto;
+  cursor: pointer;
+  text-align: center;
+  margin: 0;
+  margin-top: 0.25em;
+  margin-right: 0.25em;
+  padding: var(--fides-consent-banner-button-padding);
+  background: var(--fides-consent-banner-button-background);
+  color: var(--fides-consent-banner-button-text-color);
+  border: none;
+  border-radius: var(--fides-consent-banner-button-border-radius);
+
+  font-family: inherit;
+  font-size: 100%;
+  line-height: 1.15;
+  text-decoration: none;
+}
+
+button.fides-consent-banner-button:hover {
+  background: var(--fides-consent-banner-button-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-primary {
+  background: var(--fides-consent-banner-button-primary-background);
+  color: var(--fides-consent-banner-button-primary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-primary:hover {
+  background: var(--fides-consent-banner-button-primary-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-secondary {
+  background: var(--fides-consent-banner-button-secondary-background);
+  color: var(--fides-consent-banner-button-secondary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-secondary:hover {
+  background: var(--fides-consent-banner-button-secondary-hover-background);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary {
+  background: var(--fides-consent-banner-button-tertiary-background);
+  color: var(--fides-consent-banner-button-tertiary-text-color);
+}
+
+button.fides-consent-banner-button.fides-consent-banner-button-tertiary:hover {
+  background: var(--fides-consent-banner-button-tertiary-hover-background);
+}
+`
+
+/**
+ * Builds a 'style' element containing the CSS styles for the consent banner
+ */
+const buildStyles = (): Node => {
+  const style = document.createElement("style");
+  style.innerHTML = globalBannerStyle;
+  return style;
+};
+
+/**
+ * Initialize the Fides Consent Banner, with optional extraOptions to override defaults.
+ * 
+ * (see the type definition of ConsentBannerOptions for what options are available)
+ */
+export const initBanner = async (defaults: CookieKeyConsent, extraOptions?: ConsentBannerOptions): Promise<void> => {
+  // If the user provides any extra options, override the defaults
+  debugLog("Initializing Fides consent banner with consent cookie defaults...", defaults);
+  if (extraOptions !== undefined) {
+    if (typeof extraOptions !== "object") {
+      // eslint-disable-next-line no-console
+      console.error("Invalid 'extraOptions' arg for Fides.banner(), ignoring", extraOptions);
+    } else {
+      setBannerOptions({ ...getBannerOptions(), ...extraOptions });
+    }
+  }
+  const options: ConsentBannerOptions = getBannerOptions();
+
+  debugLog("Validating Fides consent banner options...", options);
+  if (!validateBannerOptions(options)) {
+    return;
+  }
+
+  if (options.isDisabled) {
```

Comment:
Support a global disabled flag. This allows a user to choose to disable the banner if they know something we don't - like they want to roll their own geolocation, etc.